From: Chow Loong Jin <hyperair@debian.org>
Date: Mon, 9 Feb 2026 16:10:44 +0800
Subject: Add debian-abi-helper to main CMakeLists.txt

We need to build `debian/dumpabiver.c` linked against `hyprland_lib` which only
exists virtually within the upstream CMakeLists.txt, so patch the dumpabiver
target directly into the main `CMakeLists.txt` and have it built together.

`dumpabiver` is a Debian-specific helper used to generate ${hyprland:Provides}
to generate abi-specific virtual package names for use with hyprland-plugins.
The package names will be saved into a shell-includable snippet at
`/usr/lib/${DEB_HOST_MULTIARCH}/hyprland/debian-abi-helper`.

Forwarded: not-needed
---
 CMakeLists.txt | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d526aa6..5dd21f1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -298,6 +298,33 @@ add_library(hyprland_lib STATIC ${SRCFILES})
 add_executable(Hyprland src/main.cpp ${TRACY_CPP_FILES})
 target_link_libraries(Hyprland hyprland_lib)
 
+# debian dumpabiver helper
+pkg_check_modules(
+  dumpabiver_deps
+  REQUIRED
+  IMPORTED_TARGET GLOBAL
+  libcrypto
+)
+add_executable(dumpabiver debian/dumpabiver.cpp)
+target_include_directories(dumpabiver PUBLIC ${dumpabiver_deps_INCLUDE_DIRS})
+target_link_libraries(dumpabiver hyprland_lib PkgConfig::dumpabiver_deps)
+
+add_custom_command(
+  OUTPUT debian-abi-helper
+  COMMAND dumpabiver > debian-abi-helper
+  DEPENDS dumpabiver
+)
+add_custom_target(
+  generate_abihelper
+  ALL
+  DEPENDS debian-abi-helper
+)
+install(
+  FILES ${CMAKE_BINARY_DIR}/debian-abi-helper
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/hyprland
+)
+# end dumpabiver stuff
+
 target_include_directories(hyprland_lib PUBLIC ${deps_INCLUDE_DIRS} ${LUA_INCLUDE_DIRS})
 target_include_directories(Hyprland PUBLIC ${deps_INCLUDE_DIRS} ${LUA_INCLUDE_DIRS})
 
